### DISCUSSION TITLE
Install open-mpi with --build-from-source failed: Bus-error: 10

### DISCUSSION THREAD

**Discussion by XushanLu** [Homebrew/discussions#1751] [Everyday usage]
I am trying to install OpenMPI using Hoembrew. The default installation comes with the gcc provided by Mac OS rather than GNU gcc. So, I tried to install it with the flag --build-from-source as suggested [here](https://stackoverflow.com/a/28989010).

However, the installation failed with the following error:
```
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS
PASS: opal_bitmap
PASS: opal_hash_table
PASS: ompi_rb_tree
../../config/test-driver: line 107: 94237 Bus error: 10           "$@" > $log_file 2>&1
FAIL: opal_proc_table
PASS: opal_tree
PASS: opal_pointer_array
PASS: opal_value_array
PASS: opal_list
PASS: opal_lifo
PASS: opal_fifo
```
Here is what I got from 'brew gist-logs open-mpi': https://gist.github.com/fe1c67d47dad3836581034baeeca7c50

Does anyone know what is causing this? Thanks very much!

**Comment by SMillerDev** [comment]
> The default installation comes with the gcc provided by Mac OS rather than GNU gcc. 

MacOS comes with LLVM instead of GCC.

> So, I tried to install it with the flag --build-from-source as suggested here.

That answer won't likely work anymore. It's 4 years old.

---

What are you trying to do? If you're trying to build OpenMPI with GNU GCC in Homebrew, that won't work and isn't supported. If you're trying to just install OpenMPI, why not just use the prebuild version?

**Reply by XushanLu** [reply]
Thanks for the reply.

I was trying to get OpenMPI installed with gnu-gcc rather than Mac gcc because that version does not support -fopenmp flag and I am suspecting it won't enable OpenMP+MPI hybrid parallelism?

Or, Mac gcc also supports OpenMP but with a different flag?
